x-pack/legacy/plugins/ml/server/lib/ml_telemetry/make_ml_usage_collector.ts

```diff
@@ -28,11 +26,10 @@ export function makeMlUsageCollector(
     isReady: () => true,
     fetch: async (): Promise<MlTelemetry> => {
       try {
-        const savedObjectsClient = getSavedObjectsClient(elasticsearchPlugin, savedObjects);
-        const mlTelemetrySavedObject = (await savedObjectsClient.get(
-          'ml-telemetry',
-          ML_TELEMETRY_DOC_ID
-        )) as MlTelemetrySavedObject;
+        const mlTelemetrySavedObject = ((await savedObjects
+          .createInternalRepository()
+          .get('ml-telemetry', ML_TELEMETRY_DOC_ID)) as unknown) as MlTelemetrySavedObject;
```

Comment:
Can you explain why the `as unknown` is now necessary? I wonder if [this upcoming PR related to saved objects](https://github.com/elastic/kibana/pull/58022) might allow us to do this in a better way? Is it worth waiting until it is in maybe?

Reply:
You should already be able to specify a generic for `.get()`, #58022, just makes it required. So `.get<MlTelemetrySavedObject>('ml-telemetry', ML_TELEMETRY_DOC_ID)` should work out of the box.

Reply:
Updated type in 486fc3b5cc351c937f217ab85fc9adc9fadf91a4